### PR TITLE
Allow routing table to be filtered by index pattern

### PR DESCRIPTION
**Pull request by tlrx**
Cluster state filtering does not work well for routing tables: it only matches exact index names whereas indices are fully resolved in order to filter the metadata part of the response.

This can lead to incoherent data when using an index pattern in the cluster state api:

With no index pattern:

```
GET /_cluster/state/metadata,routing_table?filter_path=**.state
{
  "metadata" : {
    "indices" : {
      "index1" : {
        "state" : "open"
      },
      "index2" : {
        "state" : "open"
      }
    }
  },
  "routing_table" : {
    "indices" : {
      "index1" : {
        "shards" : ...
      },
      "index2" : {
        "shards" : ...
      }
    }
  }
}
```

With index pattern, only metadata is returned:

```
GET /_cluster/state/metadata,routing_table/index*?filter_path=**.state
{
  "metadata" : {
    "indices" : {
      "index1" : {
        "state" : "open"
      },
      "index2" : {
        "state" : "open"
      }
    }
  }
}
```

This commit aligns the behaviour of the filtering of cluster state's routing table with the filtering of cluster state's metadata so that coherent data are returned for both routing table & metadata when index pattern is requested.

### CODE REVIEW

File: rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/20_filtering.yaml [hunk 1]
```diff
@@ -130,3 +130,34 @@ setup:
...
+        type:  type
+        id:    testing_document
+        body:
+            "text" : "The quick brown fox is brown."
+
+  - do:
+      cluster.state:
+        metric: [ routing_table, metadata ]
+        index: [ index* ]
+
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/20_filtering.yaml]: can you also remove the calls to indices.refresh that don't seem needed in this file? Maybe keep it as a separate commit though

**Comment by javanna** [comment]
LGTM thanks @tlrx for making this fix separately

File: rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/20_filtering.yaml (hunk 1 above)
Comment (tlrx) [inline rest-api-spec/src/main/resources/rest-api-spec/test/cluster.state/20_filtering.yaml]: Sure

**Comment by tlrx** [comment]
Thanks @javanna !
